src/doubledescription.jl

```diff
@@ -97,6 +97,407 @@ function doubledescription(h::HRepresentation, solver = nothing; kws...)
+        addintersection!(data, index, parent)
+    end
+    return index
+end
+
+using LinearAlgebra
+function combine(β, p1::AbstractVector, value1, p2::AbstractVector, value2)
+    λ = (value2 - β) / (value2 - value1)
+    return λ * p1 + (1 - λ) * p2
+end
+function combine(β, p::AbstractVector, pvalue, r::Polyhedra.Ray, rvalue)
+    λ = (β - pvalue) / rvalue
+    return p + λ * r
+end
+combine(β, r::Polyhedra.Ray, rvalue, p::AbstractVector, pvalue) = combine(β, p, pvalue, r, rvalue)
+function combine(β, r1::Polyhedra.Ray, value1, r2::Polyhedra.Ray, value2)
+    # should take
+    # λ = value2 / (value2 - value1)
+    @assert 0 <= value2 / (value2 - value1) <= 1
+    # By homogeneity we can avoid the division and do
+    #newr = value2 * r1 - value1 * r2
+    # but this can generate very large numbers (see JuliaPolyhedra/Polyhedra.jl#48)
+    # so we still divide
+    newr = (value2 * r1 - value1 * r2) / (value2 - value1)
+    # In CDD, it does value2 * r1 - value1 * r2 but then it normalize the ray
+    # by dividing it by its smallest nonzero entry (see dd_CreateNewRay)
+    return Polyhedra.simplify(newr)
+end
+
+function addintersection!(data, idx1, idx2)
+    if idx1.cutoff > idx2.cutoff
+        return addintersection!(data, idx2, idx1)
+    end
+    i = idx2.cutoff
+    if isin(data, i, idx1)
+        return
+    end
+    el1 = data[idx1]
+    el2 = data[idx2]
+    h = data.halfspaces[i]
+    newel = combine(h.β, el1, h.a ⋅ el1, el2, h.a ⋅ el2)
+    add_adjacent_element!(data, i - 1, newel, idx1)
+end
+function add_if_adjacent!(data, i::Integer, el1, el2)
+    # Condition (c_k) in [FP96]
+    if el1.cutoff != el2.cutoff
+        if isadjacent(data, i, el1, el2)
+            addintersection!(data, el1, el2)
+        end
+    end
+end
+
+_shift(el::AbstractVector, line::Line) = el + Polyhedra.coord(line)
+_shift(el::Line, line::Line) = el + line
+_shift(el::Ray, line::Line) = el + Polyhedra.Ray(Polyhedra.coord(line))
+function _λ_proj(r::VStruct, line::Line, h::HRepElement)
+    # Line or ray `r`:
+    # (r + λ * line) ⋅ h.a == 0
+    # λ = -r ⋅ h.a / (line ⋅ h.a)
+    return -r ⋅ h.a / (line ⋅ h.a)
+end
+function _λ_proj(x::AbstractVector, line::Line, h::HRepElement)
+    # Point `x`:
+    # (x + λ * line) ⋅ h.a == h.β
+    # λ = (h.β - x ⋅ h.a) / (line ⋅ h.a)
+    return (h.β - x ⋅ h.a) / (line ⋅ h.a)
+end
+function line_project(el, line, h)
+    λ = _λ_proj(el, line, h)
+    return Polyhedra.simplify(_shift(el, λ * line))
+end
+function hline(data, line::Line, i, h)
+    value = h.a ⋅ line
+    if !Polyhedra.isapproxzero(value)
+        if data.cutline[i] === nothing
+            # Make `lineray` point inward
+            data.cutline[i] = value > 0 ? -line : line
+            return true, line
+        else
+            line = line_project(line, data.cutline[i], h)
+        end
+    end
+    data.nlines[i] += 1
+    return false, line
+end
+# TODO remove solver arg
+function doubledescription(hr::HRepresentation, solver = nothing)
```

Comment:
If you don't need solver and still need a second argument, this can highlight the value is ignored
```suggestion
function doubledescription(hr::HRepresentation, _)
```

Reply:
But then the user is forced to give a second argument ?

Reply:
Right sorry, just updated the suggestion:
```julia
julia> f(x, _ = nothing) = x
f (generic function with 2 methods)

julia> f(3)
3

julia> f(3, 2)
3
```

Reply:
Thanks, this is done now